test/dialect/mssql/test_engine.py

```diff
@@ -93,6 +93,27 @@ def go():
             connection,
         )
 
+    def test_pyodbc_uppercase_driver_keyword_no_warn(self):
+        dialect = pyodbc.dialect()
+        u = url.make_url("mssql://username:password@hostspec/database"
+                         "?DRIVER=ODBC Driver 17 for SQL Server")
```

Comment:
I'm -1 on this because at https://docs.sqlalchemy.org/en/13/dialects/mssql.html#hostname-connections it is documented as "driver".  As discussed in #4576, this is not a straight pass-through, as we are adding brackets before sending to pyodbc, so the user can't treat this keyword like a pyodbc key.